scripts/mcp/DiscoveryAgent/Rich/discover_cli.py

```
@@ -0,0 +1,577 @@
+    prun.add_argument("--tasks-per-iter", type=int, default=3)
+    prun.set_defaults(func=cmd_run)
+
+    return p
+
+def main():
+    args = build_parser().parse_args()
+    try:
+        asyncio.run(args.func(args))
+    except KeyboardInterrupt:
+        Console().print("\\n[yellow]Interrupted[/yellow]")
+        raise SystemExit(130)
```

Ruff (0.14.11)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)